samples/client/petstore/python/petstore_api/model_utils.py

```diff
@@ -596,8 +596,17 @@ def __setitem__(self, name, value):
                 )
         # attribute must be set on self and composed instances
         self.set_attribute(name, value)
+        found_intersection = False
         for model_instance in self._composed_instances:
-            setattr(model_instance, name, value)
+            intersection = find_args_intersection(model_instance.__class__, {name: value})
+            if len(intersection):
+                setattr(model_instance, name, value)
+                found_intersection = True
+        if not found_intersection:
+            # we didn't find an intersection for the property, it could be an
+            # additional type, so pass it through
+            for model_instance in self._composed_instances:
+                setattr(model_instance, name, value)
```

Comment:
Our current code has the same properties applied to all model instances which is consistent with openapi schemas. Why are you only assigning properties here if there are no intersections?